### PR TITLE
top level export for MaterializeResult

### PR DESCRIPTION
**Pull request by alangenfeld**
its marked as @experimental already

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16503** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16503" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16503?utm_source=stack-comment).

**Comment by jamiedemaria** [comment]
Can you add to sphinx so it appears in API docs?

File: docs/sphinx/sections/api/apidocs/assets.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ An asset is an object in persistent storage, such as a table, file, or persisted
 
 .. autodecorator:: asset
 
+.. autoclass:: MaterializeResult
```
Comment (alangenfeld) [inline docs/sphinx/sections/api/apidocs/assets.rst:10]: sort of agressive to put it up top here, but a lot of the other entries in this list seem more edge case

Comment (jamiedemaria) [reply docs/sphinx/sections/api/apidocs/assets.rst:10]: i agree, but i think it's fine

**Review by jamiedemaria — APPROVED** [review]

**Pull request merged**